repo/listing.go

```diff
@@ -376,6 +378,59 @@ func ExtractIDFromListing(data []byte) (*pb.ID, error) {
 	return vendorPlay, nil
 }
 
+// Normalize converts legacy schema listing data from other users on the network
+// to fit the latest schema for consumption via the API for local use. NOTE: Legacy
+// nodes do not understand the latest schema. As such, normalized listings must not
+// be used as part of the RicardianContract and must be serialized and used as they
+// were provided by the originating node.
+func (l *Listing) Normalize() (*Listing, error) {
+	if l == nil {
+		return nil, errors.New("nil listing cannot be normalized")
+	}
+
+	if l.GetVersion() == ListingVersion {
+		return l, nil
+	}
+
+	nl, err := NewListingFromProtobuf(l.listingProto)
+	if err != nil {
+		return nil, fmt.Errorf("creating listing clone: %s", err.Error())
+	}
+
+	nlp := nl.GetProtobuf()
+	nlp.Metadata.Version = ListingVersion
+
+	if p, err := l.GetPrice(); err != nil {
+		return nil, fmt.Errorf("get price: %s", err.Error())
+	} else {
```

Comment:
This is just stylistic but generally this is an unnecessary else block. It seems more common to see:
```
p, err := l.GetPrice()
if err != nil {
    return err
}
npb.Item.BigPrice = p.Amount.String()